mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -376,13 +376,17 @@ createGlobalLoadLoop(PatternRewriter &b, Location loc, GpuAllocOp loadBuffer,
   Type loadType = vectorTypeOrSelf(elementType, vectorLen);
   Value zero = b.createOrFold<arith::ConstantIndexOp>(loc, 0);
 
-  SmallVector<Value, 4> globalStart = {zero, bid, tid, zero};
-  SmallVector<Value, 4> vectorStartOuter(4, zero);
+  Value g_block, m_block, n_block;
+  std::tie(g_block, m_block, n_block) =
+      layout::gridLayout(b, loc, bid, upperBounds[2], upperBounds[3], numCU);
+  SmallVector<Value, 4> globalStart = {zero,    g_block, m_block,
+                                       n_block, tid,     zero};
+  SmallVector<Value, 4> vectorStartOuter(6, zero);
```

Comment:
This could all just be a threadwise_read_into... 
i.e. no need to repeat the logic